addons/point_of_sale/static/tests/unit/render_service.test.js

```diff
@@ -0,0 +1,44 @@
+import { expect, mountOnFixture, test } from "@odoo/hoot";
+import { getTemplate } from "@web/core/templates";
+import { Component, useState, xml } from "@odoo/owl";
+import { MainComponentsContainer } from "@web/core/main_components_container";
+import { useService } from "@web/core/utils/hooks";
+import { registry } from "@web/core/registry";
+import { makeEnv, startServices } from "@web/env";
+
+test("test the render service", async () => {
+    class ParentComponent extends Component {
+        static props = [];
+        static components = { MainComponentsContainer };
+        static template = xml`
+            <MainComponentsContainer />
+        `;
+        setup() {
+            useService("renderer");
+        }
+    }
+
+    class ComponentToBeRendered extends Component {
+        static props = [];
+        static template = xml`
+            <div> It's me, Mario! </div>
+        `;
+        setup() {
+            this.state = useState({ number: 1 });
+        }
```

Comment:
Instead of an unused state, how about introducing props? Then pass a props to the `toHtml` call.